lib/services.js

```diff
@@ -191,6 +207,26 @@ export default {
                  });
     },
 
+   /**
+    * Check that hashedStream.completedHash matches header contentMd5.
+    * @param {object} request - request linked to the stream containine the data
+    * @param {string} hashedStream - hased stream once completed
+    * {key, dataStoreName (mem/file)}
+    * @param {RequestLogger} log - the current request logger
+    * @return {object} errors
+    */
+    checkHashMatchMD5(request, hashedStream, log) {
+        if (request.contentMD5 && hashedStream.completedHash &&
+        request.contentMD5 !== hashedStream.completedHash) {
+            log.debug('contentMD5 and completedHash does not match',
+            { method: 'checkHashMatchMD5',
+            completedHash: hashedStream.completedHash,
+            contentMD5: request.contentMD5 });
+            return errors.BadDigest;
+        }
+        return {};
```

Comment:
i think it would make sense for this function to return true if there is a match and false if not.  Then you can set the errors.BadDigest if you get false.